File: admin/src/types/product.ts
export interface Product {
  id: string
  name: string
  slug: string
  sku: string
  category: string
  price: number
  discount_price: number
  short_description: string
  description: string
  stock_quantity: number
  main_image: string
  gallery_images: string[]
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface ProductFormData {
  name: string
  category: string
  price: number
  discount_price: number
  short_description: string
  description: string
  stock_quantity: number
  is_active: boolean
  main_image?: File
}

File: backend/models/product.go
package models

type Product struct {
	ID               string   `json:"id"`
	Name             string   `json:"name"`
	Slug             string   `json:"slug"`
	SKU              string   `json:"sku"`
	Category         string   `json:"category"`
	Price            float64  `json:"price"`
	DiscountPrice    float64  `json:"discount_price"`
	ShortDescription string   `json:"short_description"`
	Description      string   `json:"description"`
	StockQuantity    int      `json:"stock_quantity"`
	CostPrice        float64  `json:"cost_price"`
	MainImage        string   `json:"main_image"`
	GalleryImages    []string `json:"gallery_images"`
	IsActive         bool     `json:"is_active"`
	CreatedAt        string   `json:"created_at"`
	UpdatedAt        string   `json:"updated_at"`
}
